ElectronClient/app.js

```diff
@@ -599,7 +599,13 @@ class Application extends BaseApplication {
 			label: _('Refresh templates'),
 			click: async () => {
 				const templates = await TemplateUtils.loadTemplates(Setting.value('templateDir'));
-				if (shim.isElectron()) Setting.setValue('availableTemplates', templates);
+				Setting.setValue('availableTemplates', templates);
+
+				setTimeout(() => {
+					if (!templates.filter(i => i.value == Setting.value('defaultNote')).length) Setting.setValue('defaultNote', '0');
+					if (!templates.filter(i => i.value == Setting.value('defaultTodo')).length) Setting.setValue('defaultTodo', '0');
+				}, 2000);
```

Comment:
This appears twice. Would making it into a function inside templateUtils be ok?

Reply:
Actually I think it should be pushed into MainScren.jsx That way it will happen right away when a user deletes a template.

Reply:
Also I think [Array.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) Might be a bit cleaner.

Reply:
Ah, didn't know about Array.find. I'll try it out. 

I've tried putting this test inside Mainscreen createNewNote,. Problem is it seems to need a timeout of about 1-2 sec for the templates array not to turn up blank (which corrupts the test), and i want the test to determine if template should be null or not. I guess just moving the current tests to createNewNote would work. Drawback is the deleted template would still appear on the first new note/todo and then not.

Reply:
That 1-2 sec is really weird, and I suspect it won't be needed if you move the code to createNewNote. Keep in mind that you can access the templates in MainScreen by using
`this.props.templates`

Reply:
Still not sure why it only worked with timeout at first. I might have messed up syntax when testing without or something. Anyway, seems to work as expected when put in createNewNote now!